kratos/processes/assign_scalar_variable_to_entities_process.h

```diff
@@ -0,0 +1,292 @@
+     */
+    void ExecuteInitializeSolutionStep() override
+    {
+        Execute();
+    }
+
+    ///@}
+    ///@name Access
+    ///@{
+
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override
+    {
+        return "AssignScalarVariableToEntitiesProcess";
+    }
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const override
+    {
+        rOStream << "AssignScalarVariableToEntitiesProcess";
+    }
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const override
+    {
+    }
+
+    ///@}
+    ///@name Friends
+    ///@{
+    ///@}
+
+protected:
+
+    ///@name Protected static Member Variables
+    ///@{
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+    /// Copy constructor.
+    AssignScalarVariableToEntitiesProcess(AssignScalarVariableToEntitiesProcess const& rOther);
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+    ///@}
+    ///@name Protected  Access
+    ///@{
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+    ///@}
+
+private:
+
+    ///@name Static Member Variables
+    ///@{
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    ModelPart& mrModelPart;     /// The model part where to assign the values
+    std::string mVariableName;  /// The name of the variable
+    double mDoubleValue;        /// The double value to assign
+    int mIntValue;              /// The integer value to assign
+    bool mBoolValue;            /// The boolean value to assign
+    std::size_t mMeshId;        /// The mesh id
+
+    ///@}
+    ///@name Private Operators
+    ///@{
+
+    /**
+     * @brief This method assigns the value (with OMP)
+     * @param rVar The variable to be assigned
+     * @param Value The value to assign
+     */
+    template< class TVarType, class TDataType >
+    void InternalAssignValue(TVarType& rVar, const TDataType Value)
+    {
+        auto& r_entities_array = GetEntitiesContainer();
+        const int number_of_entities = static_cast<int>(r_entities_array.size());
+
+        if(number_of_entities != 0) {
+            const auto it_begin = r_entities_array.begin();
+
+            #pragma omp parallel for
+            for(int i = 0; i<number_of_entities; i++) {
+                auto it_entity = it_begin + i;
+
+                it_entity->SetValue(rVar, Value);
+            }
+        }
+    }
+    
+    /**
+     * @brief This method assigns the value (without OMP)
+     * @param rVar The variable to be assigned
+     * @param Value The value to assign
+     */
+    template< class TVarType, class TDataType >
+    void InternalAssignValueSerial(TVarType& rVar, const TDataType Value)
```

Comment:
I don't see the reason this is needed, it is only used for  array3-component-variables but I don't understand why

Maybe some leftover that should be removed

Reply:
I don't' know but the template argument doesn't harm

Reply:
I meant this 
...

Reply:
answer see below